topology: add cml support with da7219 and max98357a

macchian opened the pull request:
Reuse sof-glk-da7219-kwd by included cml.m4
for sof-cml-da7219-max98357a topology with KWD pipeline enabling

Signed-of-by: Mac Chiang <mac.chiang@intel.com>

lgirdwood commented on tools/topology/sof-glk-da7219-kwd.m4:
```diff
@@ -200,44 +212,70 @@ SectionGraph."pipe-sof-glk-keyword-detect" {
 #
 
+ifelse(PLATFORM, `cml',
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `1',
```
This conditional logic should also be in platform.m4, it looks like the only delat is SSP bclk and fsync

macchian replied:
@lgirdwood ok, eventually I moved all cml specific into platform.m4.

lgirdwood commented on tools/topology/platform/intel/cml.m4:
```diff
@@ -11,3 +12,75 @@ undefine(`SSP_MCLK_RATE')
+undefine(`HDMI2_INDEX')
+define(`HDMI2_INDEX', `2')
+
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `1',
```
Why do we have multiple checks for ABI version ? Sure we just need one check.

macchian replied:
Thanks for pointing out. Yes, one check requires only.

macchian commented:
@jajanusz, could you help +1 approval or someone else? Thanks.

lgirdwood commented:
SOFCI TEST

lgirdwood commented:
Jenkins CI is known issues.

macchian commented:
> Jenkins CI is known issues.

Thanks @lgirdwood , @jajanusz . I am not authorized to merge this PR. Could you help this? Thanks.

lgirdwood commented:
SOFCI TEST

plbossart commented on tools/topology/platform/intel/bxt.m4:
```diff
@@ -1,6 +1,73 @@
+define(`SSP_INDEX', 2)
+
+undefine(`SSP_NAME')
+define(`SSP_NAME', `SSP2-Codec')
```
it makes no sense to me to add platform-specific definitions in a file that's related to the SoC architecture.
e.g. for Up2 the SSP used is 5, not 2.

Good idea to use macros, but bad location for those macros.

macchian replied:
> it makes no sense to me to add platform-specific definitions in a file that's related to the SoC architecture.
> e.g. for Up2 the SSP used is 5, not 2.
> 
> Good idea to use macros, but bad location for those macros.

@plbossart , not sure if I understand correctly, do you mean these two marcos should be located separately in glk.m4 corresponding to SSP2 and bxt.m4 corresponding to SSP5?

plbossart replied:
glk.m4 and bxt.m4 are SOC definitions. You should not add machine driver specific values in there. That should be e.g. glk_da7219.m4 or something .

lgirdwood replied:
@macchian keep the SSP/DMIC physical port IDs in the machine specific topology file but the working SSP/DMIC configs for the SoC can stay in the SoC file (i.e. bxt.m4 and glk.m4)

macchian replied:
@lgirdwood , @plbossart , in case I misunderstood your suggestions.  I posted the logic below for your commenting if anymore.
1. remove SSP_INDEX, SSP_NAME, DMIC_PCM_NUM in bxt.m4
2. create glk_da7219.m4
    added glk corresponding settings SSP_INDEX, SSP_NAME,DMIC_PCM_NUM
3. create cml_da7219.m4
    added cml corresponding settings SSP_INDEX, SSP_NAME,DMIC_PCM_NUM
4. included platform_da7219 in sof-glk-da7219-kwd.m4

lgirdwood replied:
@macchian ok, sounds correct, but can you push it for us to confirm. Thanks !

macchian replied:
> @macchian ok, sounds correct, but can you push it for us to confirm. Thanks !

@plbossart , @lgirdwood , could you check the latest update, any comments?

macchian commented:
> Yep, I think that looks right - you have DMIC, SSP mappings in board file and good SSP configs in the SoC file.

Thanks @lgirdwood ,  since I'm still not able to pulled it, could you help merge this if no fix requires?

lgirdwood commented:
SOFCI TEST

lgirdwood commented:
@macchian lets re-run CI since it had some fixes recently.

lgirdwood commented:
@macchian can you check the CI, seems we have a lot of fail reports on CML

macchian commented:
> @macchian can you check the CI, seems we have a lot of fail reports on CML

yes, I rebased the code and fixed it. Checking the CI later when it reports.

macchian commented:
> > @macchian can you check the CI, seems we have a lot of fail reports on CML
> 
> yes, I rebased the code and fixed it. Checking the CI later when it reports.

@lgirdwood , I am not able to login the QuickBuild system, could you point out the new failed logs below?

Internal Intel CI System/merge/codecheck — Bugs were found
sof-ci/jenkins/pr-device-test — On-device Test Failed:GLK_BOB_DA7219 APL_UP2_NOCODEC

lgirdwood commented:
CI jenkins known issues.
